lib/natalie/compiler/pass0c.rb

```diff
@@ -24,7 +25,62 @@ def process_iter(exp)
         )
       end
 
+      def process_defined(exp, block_with_args=nil)
+        @in_defined_statement = true
+        exp = exp.new(*exp.map {|sub_exp| sub_exp.is_a?(Sexp) ? process(sub_exp) : sub_exp})
```

Comment:
The BasePass has `process_atom` which should work:

```suggestion
        exp = exp.new(*exp.map { |e| process_atom(e) })
```

Reply:
This pattern is repeated in a few places btw.